core/raftlease/fsm.go

```diff
@@ -258,15 +284,20 @@ func (f *FSM) Snapshot() (raft.FSMSnapshot, error) {
 		}
 	}
 
-	pinned := make([]SnapshotKey, len(f.pinned))
-	i := 0
-	for key := range f.pinned {
-		pinned[i] = SnapshotKey{
+	pinned := make(map[SnapshotKey][]string)
+	for key, tags := range f.pinned {
+		if tags.IsEmpty() {
+			continue
+		}
+		entities := make([]string, tags.Size())
+		for i, t := range tags.Values() {
```

Comment:
Do we want to use tags.Values() that comes out in random order, or tags.SortedValues() that comes out in a stable order?

Alternatively we can use Values() and then call
strings.Sort(entities) at the end.